crates/iceberg/src/lib.rs

```diff
@@ -18,6 +18,7 @@
 //! Native Rust implementation of Apache Iceberg
 
 #![deny(missing_docs)]
+#![allow(dead_code)]
```

Comment:
I prefer to remove this.

Reply:
This is to avoid clippy warning.

Reply:
Adding this at the library level may result in unexpected dead code within our library. Instead, we could consider adding it at the module level, such as `avro`. Additionally, we should ensure that these additions are eventually removed.